### PR TITLE
Support Multiple RDF Types

### PR DESCRIPTION
**Pull request by terrellt**
This also does some abstraction work to try and make it easy to add other terms that support multiple values, or change how type is done.

### CODE REVIEW

**Comment by terrellt** [comment]
I'm not sure why I had to adjust the context spec.

File: spec/pragmatic_context_spec.rb [hunk 1]
```diff
@@ -43,7 +43,10 @@ class DummyResource
   end
 
   it 'should have contexts' do
-    expect(subject.as_jsonld['@context'].keys).to eq ["license", "title"]
+    keys = subject.as_jsonld['@context'].keys
```
Comment (no-reply) [inline spec/pragmatic_context_spec.rb]: maybe `expect(keys).to contain_exactly('licence', 'title')` ?

**Comment by no-reply** [comment]
I'm not sure either, but the adjusted spec seems like an improvement.

**Comment by no-reply** [comment]
Can I nag you for some Yard docs?  Especially, I'd like to see documentation on the method level for `Configuration` and class/module level documentation on `Reflection` and its friends.

**Comment by terrellt** [comment]
@no-reply Those YARD-docs good? Been quite a while since I've made them.

**Comment by no-reply** [comment]
Yeah, that looks good. Thanks.  I think I'm good with this. :+1:

**Pull request merged**